DeployableByOLM  panics when reaching time out checking for Subscription.status.installedCSV

## Bug Description
The DeployableByOLM will throw a stack trace while checking the Subscription.status.installedCSV does not reach the desired state in time. 

I _believe_ this may be related to the go function checking for the readiness in the subscription not halting once we reach timeouts, or timeouts not propagating cancellation across timeouts.

## Version and Command Invocation

Git commit hash: d6389c597717aabbce381ab6a9c2047717d781eb

## Steps to Reproduce:
I'm not sure how to reliable reproduce without changing the code to reduce timeouts because these are not configurables. That said, see the below log for the output once we got into the DeployableByOLM check[1]

## Expected Result
The check should at the very least return an error (vs. failed) if it's unable to complete its execution.


## Actual Result
The tool threw a stack trace, so no results were populated at all.


## Additional Context
[1]
```
time="2021-09-08T04:24:19Z" level=debug msg="Creating CatalogSource kogito-operator in namespace openshift-marketplace"
time="2021-09-08T04:24:19Z" level=debug msg="CatalogSource kogito-operator is created successfully in namespace openshift-marketplace"
time="2021-09-08T04:24:22Z" level=debug msg="Creating OperatorGroup kogito-operator in namespace kogito-operator"
time="2021-09-08T04:24:22Z" level=debug msg="OperatorGroup kogito-operator is created successfully in namespace kogito-operator"
time="2021-09-08T04:24:25Z" level=debug msg="Creating Subscription kogito-operator in namespace kogito-operator"
time="2021-09-08T04:24:25Z" level=debug msg="Subscription kogito-operator is created successfully in namespace kogito-operator"
time="2021-09-08T04:24:27Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:24:28.575629      14 request.go:668] Waited for 1.005491448s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/monitoring.coreos.com/v1?timeout=32s
time="2021-09-08T04:24:30Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:24:30Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:24:32Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:24:35Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:24:35Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:24:37Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:24:38.580576      14 request.go:668] Waited for 1.305845497s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/coordination.k8s.io/v1?timeout=32s
time="2021-09-08T04:24:40Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:24:40Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:24:42Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:24:44Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:24:44Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:24:46Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:24:48.591908      14 request.go:668] Waited for 1.604564062s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/coordination.k8s.io/v1?timeout=32s
time="2021-09-08T04:24:49Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:24:49Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:24:51Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:24:54Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:24:54Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:24:56Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:24:58.593979      14 request.go:668] Waited for 1.905470024s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/discovery.k8s.io/v1?timeout=32s
time="2021-09-08T04:24:59Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:24:59Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:01Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:25:04Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:04Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:06Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:25:08.602199      14 request.go:668] Waited for 2.206066173s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/metal3.io/v1alpha1?timeout=32s
time="2021-09-08T04:25:09Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:09Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:11Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:25:14Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:14Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:16Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:25:18.615602      14 request.go:668] Waited for 2.504311743s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/project.openshift.io/v1?timeout=32s
time="2021-09-08T04:25:18Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:18Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:20Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:25:23Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:23Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:25Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:25:28Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:28Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:30Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:25:31.681454      14 request.go:668] Waited for 1.005566381s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/monitoring.coreos.com/v1alpha1?timeout=32s
time="2021-09-08T04:25:33Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:33Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:35Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:25:38Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:38Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:40Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:25:41.687181      14 request.go:668] Waited for 1.305425313s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/operators.coreos.com/v1?timeout=32s
time="2021-09-08T04:25:43Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:43Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:45Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:25:48Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:48Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:50Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:25:51.691601      14 request.go:668] Waited for 1.60594341s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/coordination.k8s.io/v1?timeout=32s
time="2021-09-08T04:25:52Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:52Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:54Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:25:57Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:25:57Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:25:59Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:26:01.696514      14 request.go:668] Waited for 1.905533186s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/snapshot.storage.k8s.io/v1?timeout=32s
time="2021-09-08T04:26:02Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:02Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:04Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:26:07Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:07Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:09Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:26:11.706719      14 request.go:668] Waited for 2.205515149s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/controlplane.operator.openshift.io/v1alpha1?timeout=32s
time="2021-09-08T04:26:12Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:12Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:14Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:26:17Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:17Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:19Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:26:21.717594      14 request.go:668] Waited for 2.484823045s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/helm.openshift.io/v1beta1?timeout=32s
time="2021-09-08T04:26:21Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:22Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:24Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:26:26Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:26Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:28Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:26:31Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:31Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:33Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:26:34.798133      14 request.go:668] Waited for 1.005525947s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/operators.coreos.com/v2?timeout=32s
time="2021-09-08T04:26:36Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:36Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:38Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:26:41Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:41Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:43Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:26:44.811436      14 request.go:668] Waited for 1.305928463s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/operators.coreos.com/v1?timeout=32s
time="2021-09-08T04:26:46Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:46Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:48Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:26:51Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:51Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:53Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:26:54.811800      14 request.go:668] Waited for 1.604680094s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/console.openshift.io/v1alpha1?timeout=32s
time="2021-09-08T04:26:55Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:26:56Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:26:58Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:27:00Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:00Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:27:02Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:27:04.815208      14 request.go:668] Waited for 1.90502257s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/snapshot.storage.k8s.io/v1?timeout=32s
time="2021-09-08T04:27:05Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:05Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:27:07Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:27:10Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:10Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:27:12Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:27:14.821884      14 request.go:668] Waited for 2.205296105s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/metal3.io/v1alpha1?timeout=32s
time="2021-09-08T04:27:15Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:15Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:27:17Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:27:20Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:20Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:27:22Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
I0908 04:27:24.828282      14 request.go:668] Waited for 2.504375571s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/whereabouts.cni.cncf.io/v1alpha1?timeout=32s
time="2021-09-08T04:27:25Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:25Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:27:25Z" level=error msg="failed to fetch Subscription.status.installedCSV: context deadline exceeded"
time="2021-09-08T04:27:25Z" level=debug msg="Dumping data in artifacts/ directory"
time="2021-09-08T04:27:27Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:27:28Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:29Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:29Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
time="2021-09-08T04:27:31Z" level=debug msg="fetching catalogsource: kogito-operator"
time="2021-09-08T04:27:31Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:27:34Z" level=debug msg="fetching operatorgroup kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:34Z" level=debug msg="fetching namespace: kogito-operator"
time="2021-09-08T04:27:34Z" level=trace msg="Deleting the resources created by Check"
time="2021-09-08T04:27:34Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
time="2021-09-08T04:27:34Z" level=debug msg="Subscription.status.installedCSV is not set yet, retrying..."
I0908 04:27:35.158308      14 request.go:668] Waited for 1.006415655s due to client-side throttling, not priority and fairness, request: GET:https://my-test-cluster:6443/apis/storage.k8s.io/v1beta1?timeout=32s
time="2021-09-08T04:27:36Z" level=debug msg="Waiting for Subscription.status.installedCSV to become ready..."
time="2021-09-08T04:27:36Z" level=debug msg="Deleting Subscription kogito-operator in namespace kogito-operator"
time="2021-09-08T04:27:36Z" level=debug msg="Subscription kogito-operator is deleted successfully from namespace kogito-operator"
time="2021-09-08T04:27:39Z" level=debug msg="fetching subscription kogito-operator from namespace kogito-operator "
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x138 pc=0x16078f9]

goroutine 858 [running]:
github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/policy/operator.(*DeployableByOlmCheck).installedCSV.func1(0xc0003b1ce0, 0xc0003526c0, 0xc000048010, 0x5a, 0xc00036bba8, 0x3, 0xc00036bc20, 0xf, 0xc00036bc20, 0xf, ...)
	/go/src/preflight/certification/internal/policy/operator/deployable_by_olm.go:190 +0x179
created by github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/policy/operator.(*DeployableByOlmCheck).installedCSV
	/go/src/preflight/certification/internal/policy/operator/deployable_by_olm.go:184 +0x119

```

--- a/certification/internal/policy/operator/deployable_by_olm.go
+++ b/certification/internal/policy/operator/deployable_by_olm.go
@@ -142,14 +142,18 @@ func (p *DeployableByOlmCheck) isCSVReady(installedCSV string, operatorData Oper
 
 	log.Trace(fmt.Sprintf("Looking for csv %s in namespace %s", installedCSV, operatorData.InstallNamespace))
 
-	// query API server for CSV by name
-	ctx, cancel := context.WithTimeout(context.Background(), 90*time.Second)
-	defer cancel()
+	ctx := context.Background()
 
 	csvReadyDone := make(chan string, 1)
+	defer close(csvReadyDone)
 
-	go func() {
-		defer close(csvReadyDone)
+	contextTimeOut := make(chan error, 1)
+	defer close(contextTimeOut)
+
+	go func(parent context.Context) {
+		ctx, cancel := context.WithTimeout(parent, 90*time.Second)
+		defer cancel()
+
 		for {
 			log.Debug("Waiting for ClusterServiceVersion to become ready...")
 			csv, _ := openshiftengine.GetCSV(installedCSV, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
@@ -157,33 +161,46 @@ func (p *DeployableByOlmCheck) isCSVReady(installedCSV string, operatorData Oper
 			if csv.Status.Phase == operatorv1alpha1.CSVPhaseSucceeded {
 				log.Debug("CSV is created successfully: ", installedCSV)
 				csvReadyDone <- fmt.Sprintf("%#v", csv)
+				return
 			}
 			log.Debug("CSV is not ready yet, retrying...")
-			time.Sleep(2 * time.Second)
+
+			select {
+			case <-ctx.Done():
+				log.Error(fmt.Sprintf("failed to fetch the csv %s: ", installedCSV), ctx.Err())
+				contextTimeOut <- ctx.Err()
+				return
+			default:
+				time.Sleep(2 * time.Second)
+			}
 		}
-	}()
+	}(ctx)
 
 	select {
 	case csv := <-csvReadyDone:
 		return len(csv) > 0, nil
-	case <-ctx.Done():
-		log.Error(fmt.Sprintf("failed to fetch the csv %s: ", installedCSV), ctx.Err())
-		return false, nil
+	case err := <-contextTimeOut:
+		return false, fmt.Errorf("%w: %s", errors.ErrK8sAPICallFailed, err)
 	}
+
 }
 
 func (p *DeployableByOlmCheck) installedCSV(operatorData OperatorData, openshiftengine openshiftengine.OpenshiftEngine) (string, error) {
 
-	ctx, cancel := context.WithTimeout(context.Background(), 180*time.Second)
-	defer cancel()
+	ctx := context.Background()
 
 	installedCSVDone := make(chan string, 1)
+	defer close(installedCSVDone)
+
+	contextTimeOut := make(chan error, 1)
+	defer close(contextTimeOut)
 
 	// query API server for the installed CSV field of the created subscription
-	go func() {
-		defer close(installedCSVDone)
+	go func(parent context.Context) {
+		ctx, cancel := context.WithTimeout(parent, 180*time.Second)
+		defer cancel()
+
 		for {
-			time.Sleep(2 * time.Second)
 			log.Debug("Waiting for Subscription.status.installedCSV to become ready...")
 			subs, _ := openshiftengine.GetSubscription(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
 			installedCSV := subs.Status.InstalledCSV
@@ -191,17 +208,26 @@ func (p *DeployableByOlmCheck) installedCSV(operatorData OperatorData, openshift
 			if len(installedCSV) > 0 {
 				log.Debug(fmt.Sprintf("Subscription.status.installedCSV is %s", installedCSV))
 				installedCSVDone <- installedCSV
+				return
 			}
 			log.Debug("Subscription.status.installedCSV is not set yet, retrying...")
+
+			select {
+			case <-ctx.Done():
+				log.Error("failed to fetch Subscription.status.installedCSV: ", ctx.Err())
+				contextTimeOut <- ctx.Err()
+				return
+			default:
+				time.Sleep(2 * time.Second)
+			}
 		}
-	}()
+	}(ctx)
 
 	select {
 	case installedCSV := <-installedCSVDone:
 		return installedCSV, nil
-	case <-ctx.Done():
-		log.Error("failed to fetch Subscription.status.installedCSV: ", ctx.Err())
-		return "", errors.ErrK8sAPICallFailed
+	case err := <-contextTimeOut:
+		return "", fmt.Errorf("%w: %s", errors.ErrK8sAPICallFailed, err)
 	}
 }
 

